### ISSUE TITLE
Handle keyboard interrupts better

### ISSUE THREAD

**Issue by joshschmelzle** [WLAN-Pi/wlanpi-chat-bot#1]
Currently we have to spam control+c when running the module directly in order to stop it. We should improve this as time permits.

```
(venv) wlanpi@wlanpi-emmc-513-waves:[~/dev/wlanpi-chat-bot]: sudo venv/bin/python3 -m chatbot
INFO:TelegramAlert:We were offline, but we're back online.
^C^C


^C^C^CProcess Process-12:
Traceback (most recent call last):
  File "/usr/lib/python3.7/multiprocessing/process.py", line 285, in _bootstrap
    util._close_stdin()
  File "/usr/lib/python3.7/multiprocessing/util.py", line 387, in _close_stdin
    sys.stdin = open(fd, closefd=False)
  File "/usr/lib/python3.7/_bootlocale.py", line 33, in getpreferredencoding
    def getpreferredencoding(do_setlocale=True):
KeyboardInterrupt
^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C
```

**Comment by wifinigel** [comment]
Not sure if it helps, but Ctrl-Z seems to work OK during my testing.

**Comment by joshschmelzle** [comment]
Thanks, that seems to work better. So, it looks like the code handles SIGTSTP (Ctrl-Z) better than SIGINT (Ctrl-C).

**Comment by joshschmelzle** [comment]
This doesn't seem to be the same thing. Ctrl-Z puts the process in the background rather than stopping it. Look at the output of `jobs` to see a list items in the background, and use `fg` to resume:

```
(venv) wlanpi@wlanpi-emmc-513-waves:[~/dev/wlanpi-chat-bot]: fg
sudo venv/bin/python3 -m chatbot
ERROR:TelegramAlert:Other error occurred: ('Connection aborted.', OSError(0, 'Error'))
```

**Comment by joshschmelzle** [comment]
Fixed by 7c332d8b95b8037065ed202984587f397d9054d7. Thank you Nigel for working on this.
